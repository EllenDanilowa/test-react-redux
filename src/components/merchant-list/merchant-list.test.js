import React from 'react';
import renderer from 'react-test-renderer';
import MerchantList from './merchant-list';
import {mount} from 'enzyme';

/* eslint-disable react/display-name, react/prop-types */
jest.mock('react-paginate', () => ({pageCount}) => (<div><p id="pageCount">&apos;{pageCount}&apos;</p></div>));
jest.mock('../merchant/merchant', () => ({item}) => (<div>merchant: &apos;{item.firstname} {item.lastname}&apos;</div>));
/* eslint-enable react/display-name, react/prop-types */

describe('MerchantList', () => {
  let element;
  let count;
  let deleteMerchant;
  let updateVisibleMerchants;
  let merchants;

  const createElement = (props, isSnapshot = true) => {
    const template = (<MerchantList {...props}/>);

    return isSnapshot ?
      renderer.create(template).toJSON() :
      mount(template);
  };

  beforeEach(() => {
    count = 2;
    deleteMerchant = () => {};
    updateVisibleMerchants = jest.fn();
    merchants = [
      {
        id: '1',
        firstname: 'name',
        lastname: 'lastname',
        avatarUrl: '',
        email: 'test@test.com',
        phone: '123456789',
        hasPremium: false,
        bids: []
      },
      {
        id: '2',
        firstname: 'name 2',
        lastname: 'lastname 2',
        avatarUrl: '',
        email: 'test2@test.com',
        phone: '123456789',
        hasPremium: true,
        bids: []
      }
    ];

    jest.spyOn(React, 'useEffect').mockImplementation((cb) => cb());
  });

  it('updates visible merchants on mounting', () => {
    element = createElement({deleteMerchant, updateVisibleMerchants, merchants, count});

    expect(updateVisibleMerchants).toHaveBeenCalled();
  });

  it('renders component', () => {
    element = createElement({deleteMerchant, updateVisibleMerchants, merchants, count});

    expect(element).toMatchSnapshot();
  });

  it('hides pagination and list of merchants if no merchants', () => {
    count = 0;
    element = createElement({deleteMerchant,updateVisibleMerchants, count});

    expect(element).toMatchSnapshot();
  });
});
