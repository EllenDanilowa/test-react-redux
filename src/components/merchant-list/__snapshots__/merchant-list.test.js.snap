// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MerchantList hides pagination and list of merchants if no merchants 1`] = `<div />`;

exports[`MerchantList renders component 1`] = `
<div>
  <div>
    merchant: '
    name
     
    lastname
    '
  </div>
  <div>
    merchant: '
    name 2
     
    lastname 2
    '
  </div>
  <div
    className="sc-AxjAm kSsZsT"
  >
    <div>
      <p
        id="pageCount"
      >
        '
        2
        '
      </p>
      <button
        id="pageChange"
        onClick={[Function]}
      />
    </div>
  </div>
</div>
`;
